staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -828,8 +838,12 @@ func (meal *cfgMeal) finishQueueSetReconfigsLocked() {
 		if limited := plState.pl.Spec.Limited; limited != nil {
 			if qCfg := limited.LimitResponse.Queuing; qCfg != nil {
 				meal.maxWaitingRequests += int(qCfg.Queues * qCfg.QueueLengthLimit)
+
+				handSize := float64(plState.pl.Spec.Limited.LimitResponse.Queuing.HandSize)
+				plState.maxSeats = uint64(math.Min(math.Ceil(float64(concurrencyLimit)*float64(priorityLevelMaxSeatsPercent)/100), math.Floor(float64(concurrencyLimit)/handSize)))
```

Comment:
(1) Why define `priorityLevelMaxSeatsPercent` as a constant integer and always convert to a float and divide by 100 instead of just defining a constant float holding the desired fraction?

(2) For the concurrencyLimit/handSize part, why convert both from integers to floats, divide, then floor when integer division does the floor?

(3) Don't we want `max(1, ...)` around all of that?

Reply:
Fixed